### PR TITLE
Common CLI environment

### PR DESCRIPTION
**Pull request by jamesr73**
Creates a single instance of UI and ProjectSettings that CLI SubCommands can share rather than each one creating it's own. Ensures settings, inc BlueprintCollection, are only loaded once.

Fixes SubCommand instantiation to make it easier to test SubCommands with mock environments.

Tidies up buildBlueprintCommands some more, including deep merge of blueprint settings.

Tweaks yargs parser to use full console width to display usage.

If this is a suitable approach to addressing this issue I can add some tests, primarily for buildBlueprintCommands, before merging. Adding a full test suite for all SubCommands and Tasks really needs separate PR(s).

### CODE REVIEW

File: src/cli/environment.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import ProjectSettings from '../models/project-settings';
+import UI from '../models/ui';
+
+let environment;
```
Comment (anithri) [inline src/cli/environment.js]: closures for the win!

**Comment by anithri** [comment]
Is it ready to go?

File: src/cli/environment.js (hunk 1 above)
Comment (jamesr73) [reply src/cli/environment.js]: LOL. So nearly did that.

**Comment by jamesr73** [comment]
Could probably use a couple of tests...

**Comment by jamesr73** [comment]
Added tests for cli/environment.js. Mocking out the dependencies to test buildBlueprintCommands is proving to be a pain and I'm done for the night. I'll get back to that tomorrow.

Up to you whether you want to merge now or wait for more tests.

**Comment by anithri** [comment]
I'm going to let you rebase it on the other merges first.

**Comment by anithri** [comment]
Rebasing it on master was pretty smooth.

see anithri/redux-cli/de7828f17dba48bf013695cc640f5f080e896102 for the result

**Comment by anithri** [comment]
Rebasing it on master was pretty smooth.

see anithri/redux-cli@de7828f17dba48bf013695cc640f5f080e896102 for the result

**Comment by jamesr73** [comment]
Good to go

**Pull request merged**